crates/red_knot_wasm/src/lib.rs

```diff
@@ -184,6 +192,30 @@ impl Workspace {
         Ok(format!("{:#?}", parsed.syntax()))
     }
 
+    pub fn format(&self, file_id: &FileHandle) -> Result<String, Error> {
+        let parsed = parsed_module(&self.db, file_id.file);
```

Comment:
I considered making `ruff_python_formatter` defining its own DB and expose a `format` query that returns `Formatted` or Unformatted(String)` but it seemed more work because it requires introducing a db trait and what not and I'm not sure if this is indeed the query API that we want. 

For now, this is ad-hoc, but it re-uses the cached AST (although we need to clone it because of API incompatibilities :()